src/redis-cli.c

```diff
@@ -2039,8 +2061,15 @@ static int issueCommandRepeat(int argc, char **argv, long repeat) {
         /* Issue the command again if we got redirected in cluster mode */
         if (config.cluster_mode && config.cluster_reissue_command) {
             /* If cliConnect fails, sleep for a while and try again. */
-            if (cliConnect(CC_FORCE) != REDIS_OK)
+            if (cliConnect(CC_FORCE) != REDIS_OK) {
                 sleep(1);
+                cliConnect(CC_FORCE);
```

Comment:
why did you add another call to cliConnect (and one that doesn't check the return value)? at the very least, we need a comment to explain it.

if i'm not mistaken, instead of that line, you may wanna add an `else` before the `if (config.cluster_pre_asking)` check below (i.e. to run it only if cliConnect succeeded).

Reply:
At first this code is
```
if (config.cluster_mode && config.cluster_reissue_command) {
    cliConnect(CC_FORCE);
```

But  https://github.com/redis/redis/pull/8884 added retry. My code has conflicts.
If failed, `config.cluster_reissue_command` is set. Then this code path will never be entered again. Because network error is rarely, so I add another retry.

Or I maybe need change the code, because I need call cliPreAsking before call real commands.

Reply:
i'm confused. i understand you had a merge conflict, do we really need another call to `cliConnect`?
p.s. IIRC 8884 didn't add a retry loop (it already existed), it only added a sleep.

Reply:
I'm wrong about 8884.
I add `cliConnection` because of this line `2049   config.cluster_reissue_command = 0;`.  If we `cliConnect` failed, then we retry success, `config.cluster_reissue_command` is unset. So we can no longer execute `cliPreAsking`.

Reply:
so maybe this is an indication that the `cliPreAsking` needs to go on the top of this function (right before `cliSendCommand` is called)?
i'm not sure. is it just me or that something doesn't feel right here?

i admit i'm a bit out of focus, trying to jump between different tasks.
please try cleaning this up, or convincing me that i'm just wrong.

p.s. i feel that `cliPreAsking` isn't a very good name. i suppose a comment above it is needed to explain what it does when when it is used. but also maybe the function can be improved?
something like `cliSendAsking`?
to me the name seemed to suggest that this function should be called before "Asking" (pre-asking). but actually it sends an ASKING before the other (real) command.

Reply:
Naming is always a different thing. What I want to mean is pre next command. But it cause misleading.
I move this part to the top.